test/e2e/util_test.go

```diff
@@ -259,6 +259,37 @@ func buildRoute(name, namespace, serviceName string) *routev1.Route {
 	}
 }
 
+// buildRouteWithHTTPHeaders returns a route definition targeting the specified service along with setting of header name and value.
+func buildRouteWithHTTPHeaders(name, namespace, serviceName string, headerName string, headerValue string) *routev1.Route {
```

Comment:
nit: Please change the name of the function so noone tries to use this expecting to be able to set request headers.
```suggestion
func buildRouteWithHTTPResponseHeaders(name, namespace, serviceName string, headerName string, headerValue string) *routev1.Route {
```

Reply:
ok